codeworld-compiler/test/Main.hs

```diff
@@ -53,6 +56,24 @@ compileErrorOutput testName = do
     errMsg <- readFile (testErrorFile testName)
     return errMsg
 
-test1 = TestCase (do err1 <- (compileErrorOutput "test1")
-                     err2 <- (compileErrorOutput "test2") 
-                     assertEqual "Test case 1" err1 err2)
+genTestCases :: [t] -> [Test]
```

Comment:
This function is just a map, with an error for the empty case.  I'd want to see this function:

    toTestCase x = x ~: do
        err1 <- compileErrorOutput x
        err2 <- savesErrorOutput x
        assertEqual x err1 err2

Then you can just map that over the list, and then add a new test to fail if the list was empty?